src/coreclr/System.Private.CoreLib/src/System/Threading/Interlocked.CoreCLR.cs

```diff
@@ -48,27 +48,60 @@ public static partial class Interlocked
         /// <returns>The original value of <paramref name="location1"/>.</returns>
         /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
         [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int Exchange(ref int location1, int value)
+        {
+#if TARGET_X86 || TARGET_AMD64 || TARGET_ARM64 || TARGET_RISCV64
+            return Exchange(ref location1, value); // Must expand intrinsic
+#else
+            if (Unsafe.IsNullRef(ref location1))
```

Comment:
wonder if it could be just `_ = location1;` and be shared with the must-expand. Or it will need a pragma to disable a warning?

Reply:
The problem with doing the null check via a dummy read is that it is an extra read of a potentially highly contended memory location. It can create up to 2x more bus traffic. I think that the explicit null check is better, given that it is costs close to nothing and the JIT is able to remove it in number of cases.